### PR TITLE
COGLayer Support for Reading and Writing

### PR DESCRIPTION
**Pull request by jbouffard**
This PR is a continuation of #680

### CODE REVIEW

File: geopyspark/geotrellis/catalog.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ def read_value(uri,
                layer_zoom,
                col,
                row,
-               zdt=None):
+               zdt=None,
+               use_cogs=False):
```
[the remark this answers is not in the document]
Comment (jbouffard) [reply geopyspark/geotrellis/catalog.py]: I honestly have not idea what happened. Some of my work is gone.

[the remark this answers is not in the document]
Comment (echeipesh) [reply geopyspark/geotrellis/catalog.py]: Rebase gone wrong ?

[the remark this answers is not in the document]
Comment (jbouffard) [reply geopyspark/geotrellis/catalog.py]: https://github.com/locationtech-labs/geopyspark/pull/692/commits/00fa3c379f0697f1dfc371fe7161443f40195fb8#diff-e5c50f070956be2d3374b19f60a8d8fbR47

[the remark this answers is not in the document]
Comment (jbouffard) [reply geopyspark/geotrellis/catalog.py]: @echeipesh Yeah, I think that's what ended up happening. It's a bit strange, though.

File: geopyspark-backend/project/Version.scala [hunk 2]
```diff
@@ -1,6 +1,6 @@
 object Version {
   val geopyspark = "0.4.2"
-  val geotrellis = "2.0.0"
+  val geotrellis = "2.1.0-SNAPSHOT"
```
Comment (echeipesh) [inline geopyspark-backend/project/Version.scala]: Can pull down the real McCoy now.

Comment (jbouffard) [reply geopyspark-backend/project/Version.scala]: :+1:

File: geopyspark/tests/base_test_class.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 class BaseTestClass(unittest.TestCase):
     if 'TRAVIS' in os.environ:
-        master_str = "local[2]"
+        master_str = "local[1]"
```
Comment (echeipesh) [inline geopyspark/tests/base_test_class.py]: Why the reduction? This could potentially hide some failures in parallelism.

Comment (jbouffard) [reply geopyspark/tests/base_test_class.py]: This was one of things I tried to see if it got Travis working. Unfortunately, it did not work. So I think that could be reverted.

**Pull request merged**